mne/event.py

```diff
@@ -827,3 +828,205 @@ def concatenate_events(events, first_samps, last_samps):
+        where t is time in samples and event_code is all events compatible with the transition
+        bitwise anded together. """
+        events_ = mne_events.copy()
+        events_[:,1] = 0
+        events_[:,2] = 0
+        for n,ev in self.events.iteritems():
+            if ev.in_use:
+                pre_ok = np.bitwise_and(ev.oldmask, mne_events[:,1]) == ev.oldbits
+                post_ok = np.bitwise_and(ev.newbits, mne_events[:,2]) == ev.newbits
+                ok_ind = np.where(pre_ok & post_ok)
+                if np.all(events_[ok_ind,2] == 0):
+                    events_[ok_ind,2] |= 1 << (n - 1)  # switch on the bit corresponding to event number
+                else:
+                    # code can handle this, but not clear if it's intended to be possible
+                    raise Exception('Multiple dacq events match trigger transition')
+        return events_
+
+    def _mne_events_to_category_times(self, cat, mne_events, sfreq):
+        """ Translate mne events to times when epochs for a given category should
+        be collected. """
+        events = self._mne_events_to_dacq(mne_events)
+        times = events[:,0]
+        refEvents_inds = np.where(events[:,2] & (1 << cat.event-1))[0]  # indices of times where ref. event occurs
+        refEvents_t = times[refEvents_inds]
+        if cat.reqevent:
+            reqEvents_inds = np.where(events[:,2] & (1 << cat.reqevent-1))[0]  # indices of times where req. event occurs
+            reqEvents_t = times[reqEvents_inds]
+            # relative (to refevent) time window (in samples) where req. event must occur (e.g. [0 200])
+            win = np.round(np.array(sorted([0, (-1)**(cat.reqwhen)*cat.reqwithin]))*sfreq)
+            refEvents_wins = refEvents_t[:,None] + win
+            req_acc = np.zeros(refEvents_inds.shape, dtype=bool)
+            for t in reqEvents_t:
+                # mark time windows where req. condition is satisfied
+                reqEvent_in_win = np.logical_and(t >= refEvents_wins[:,0], t <= refEvents_wins[:,1])
+                req_acc |= reqEvent_in_win  
+            # drop ref. events where req. event condition is not satisfied
+            refEvents_inds = refEvents_inds[np.where(req_acc)]
+            refEvents_t = times[refEvents_inds]            
+        # adjust for trigger-stimulus delay by delaying the ref. event
+        refEvents_t += np.round(self.events[cat.event].delay * sfreq)
+        return refEvents_t
+
+    def get_mne_rejection_dict(self):
+        """ Makes a mne rejection dict based on the averager parameters. Result
+        can be used e.g. with mne.Epochs """
+        return {'grad':self.megmax, 'mag':self.magmax, 'eeg':self.eegmax,
+                'eog':self.eogmax, 'ecg':self.ecgmax}
+
+    def get_epochs(self, raw, catname, picks=None, reject=None, stim_channel=None, mask=0):
+        """ Get mne.Epochs instance corresponding to the given category. """
```

Comment:
Nest the import here. In other words, put this just below this line:
```
    from .epochs import Epochs
```
Then delete the import from the top. This will solve the circular import problem you're having.